### PR TITLE
Replace ess-smart-underscore with ess-smart-equals

### PR DESCRIPTION
**Pull request by sooheon**
The former was last updated in 2011. For the benefits of the latter, see: https://github.com/genovese/ess-smart-equals

Also added R-mode to auto-mode-alist and rainbow delimiters, for convenience.

One question is how necessary you think the manual loading mechanism is, with use-package and :defer? Could I try making it work simply with use-package like other packages?

### CODE REVIEW

File: contrib/lang/ess/packages.el [hunk 1]
```diff
@@ -27,6 +27,16 @@ which require an initialization must be listed explicitly in the list.")
   ;; R --------------------------------------------------------------------------
   (eval-after-load "ess-site"
     '(progn
+       (add-to-list 'auto-mode-alist '("\\.R$" . R-mode))
+       (add-hook 'ess-mode-hook #'rainbow-delimiters-mode)
```
Comment (syl20bnr) [inline contrib/lang/ess/packages.el]: Prefer to add `rainbow-delimiters` to `ess-packages` and define `ess/init-rainbow-delimiters` where you add the hook.
Example here with python and flycheck: https://github.com/syl20bnr/spacemacs/blob/c557111fd88342216202f29312fe7ad4996f9454/contrib/lang/python/packages.el#L133-134

Comment (syl20bnr) [inline contrib/lang/ess/packages.el]: Same thing for the other mode hooks.

**Comment by syl20bnr** [comment]
I tried to defer ESS but never succeeded, if you achieved to do it, PR is very welcome :-)

File: contrib/lang/ess/packages.el (hunk 1 above)
Comment (sooheon) [inline contrib/lang/ess/packages.el]: New commit, wondering if this is better.

**Comment by syl20bnr** [comment]
Thank you ! :+1: 

Cherry-picked into develop branch.

**Pull request closed without merging**